AppCenterDataStorage/AppCenterDataStorageTests/MSTokenTests.m

```diff
@@ -21,6 +37,199 @@ - (void)tearDown {
   [super tearDown];
 }
 
-- (void)testGetToken {
+- (void)testGetTokenResult {
+
+  // When
+  MSTokenResult *result = [[MSTokenResult alloc] initWithPartition:partitionName
+                                                         dbAccount:dbAccount
+                                                            dbName:dbName
+                                                  dbCollectionName:dbCollectionName
+                                                             token:token
+                                                            status:status
+                                                         expiresOn:expiresOn];
+
+  // Then
+  XCTAssertTrue([partitionName isEqualToString:result.partition]);
+  XCTAssertTrue([token isEqualToString:result.token]);
+  XCTAssertTrue([status isEqualToString:result.status]);
+  XCTAssertTrue([dbName isEqualToString:result.dbName]);
+  XCTAssertTrue([dbCollectionName isEqualToString:result.dbCollectionName]);
+  XCTAssertTrue([expiresOn isEqualToString:result.expiresOn]);
+  XCTAssertTrue([dbAccount isEqualToString:result.dbAccount]);
+
+  // When
+  NSString *resultString = [result serializeToString];
+  NSDictionary *resultDic = [self deserializeDataString:resultString];
+
+  // Then
+  XCTAssertTrue([partitionName isEqualToString:resultDic[kMSPartition]]);
+  XCTAssertTrue([token isEqualToString:resultDic[kMSToken]]);
+  XCTAssertTrue([status isEqualToString:resultDic[kMSStatus]]);
+  XCTAssertTrue([dbName isEqualToString:resultDic[kMSDbName]]);
+  XCTAssertTrue([dbAccount isEqualToString:resultDic[kMSDbAccount]]);
+  XCTAssertTrue([dbCollectionName isEqualToString:resultDic[kMSDbCollectionName]]);
+  XCTAssertTrue([expiresOn isEqualToString:resultDic[kMSExpiresOn]]);
+}
+
+- (void)testGetTokenResultWithString {
+
+  // If
+  NSString *tokenString =
+      [NSString stringWithFormat:@"{\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\",\"%@\":\"%@\"}",
+                                 kMSPartition, partitionName, kMSToken, token, kMSStatus, status, kMSDbName, dbName, kMSDbCollectionName,
+                                 dbCollectionName, kMSExpiresOn, expiresOn, kMSDbAccount, dbAccount];
+
+  // When
+  MSTokenResult *result = [[MSTokenResult alloc] initWithString:tokenString];
+
+  // Then
+  XCTAssertTrue([partitionName isEqualToString:result.partition]);
```

Comment:
Please use `XCTAssertEqual(expected, actual);`

Reply:
@jwallra `XCTAssertEqual` check for reference of the object, not the value of the object.

@jinquan2019 The recommendation is to use `XCTAssertEqualObjects` for string comparison.

Reply:
Yes `XCTAssertEqualObjects` can be used.